yearn/apy/curve/simple.py

```diff
@@ -154,80 +165,160 @@ def simple(vault, samples: ApySamples) -> Apy:
+
+        return len(vault.strategies) == 1 and "convex" in vault.strategies[0].name.lower()
+
+    def apy(self, base_asset_price, pool_price, base_apr, pool_apy: float, management_fee: float, performance_fee: float) -> Apy:
+        """The standard APY data."""
+        apy_data = self.get_detailed_apy_data(base_asset_price, pool_price, base_apr)
+        gross_apr = (1 + (apy_data.cvx_apr * apy_data.cvx_debt_ratio)) * (1 + pool_apy) - 1
+
+        cvx_net_apr = (apy_data.cvx_apr_minus_keep_crv + apy_data.convex_reward_apr) * (1 - performance_fee) - management_fee
+        cvx_net_farmed_apy = (1 + (cvx_net_apr / COMPOUNDING)) ** COMPOUNDING - 1
+        cvx_net_apy = ((1 + cvx_net_farmed_apy) * (1 + pool_apy)) - 1
+
+        fees = ApyFees(performance=performance_fee, management=management_fee, cvx_keep_crv=apy_data.cvx_keep_crv)
+        return Apy("convex", gross_apr, cvx_net_apy, fees)
+
+    def get_detailed_apy_data(self, base_asset_price, pool_price, base_apr) -> ConvexDetailedApyData:
+        """Detailed data about the apy."""
+        cvx_keep_crv = self._cvx_strategy.keepCRV(block_identifier=self.block) / 1e4
+
+        cvx_booster = contract(addresses[chain.id]['convex_booster'])
+        cvx_fee = self._get_convex_fee(cvx_booster, self.block)
+        convex_reward_apr = self._get_reward_apr(self._cvx_strategy, cvx_booster, base_asset_price, pool_price, self.block)
+
+        cvx_boost = self._get_cvx_boost()
+        cvx_printed_as_crv = self._get_cvx_emissions_printed_as_crv()
+        cvx_apr = ((1 - cvx_fee) * cvx_boost * base_apr) * (1 + cvx_printed_as_crv) + convex_reward_apr
+        cvx_apr_minus_keep_crv = ((1 - cvx_fee) * cvx_boost * base_apr) * ((1 - cvx_keep_crv) + cvx_printed_as_crv)
+        
+        return ConvexDetailedApyData(cvx_apr, cvx_apr_minus_keep_crv, cvx_keep_crv, self._debt_ratio, convex_reward_apr)
+
+    def _get_cvx_emissions_printed_as_crv(self) -> float:
+        """The amount of CVX emissions at the current block, in terms of CRV."""
+        crv_price = magic.get_price(curve.crv, block=self.block)
+        total_cliff = 1e3 # the total number of cliffs to happen
+        max_supply = 1e2 * 1e6 * 1e18 # the maximum amount of CVX that will be minted
+        reduction_per_cliff = 1e23 # the reduction in emission per cliff
+        cvx = contract(addresses[chain.id]['cvx'])
+        supply = cvx.totalSupply(block_identifier=self.block) # current supply of CVX 
+        cliff = supply / reduction_per_cliff # the current cliff we're on
+        if supply <= max_supply:
+            reduction = total_cliff - cliff
+            cvx_minted_as_crv = reduction / total_cliff
+            cvx_price = magic.get_price(cvx, block=self.block)
+            converted_cvx = cvx_price / crv_price
+            return cvx_minted_as_crv * converted_cvx
+        else:
+            return 0
+
+    def _get_cvx_boost(self) -> float:
+        """The amount of boost being applied to this pool by Convex."""
```

Comment:
"The Curve boost (1-2.5x) being applied to this pool thanks to veCRV locked in Convex's voter proxy."